abstract_factory/abstract_factory.rb

```diff
@@ -0,0 +1,9 @@
+class AbstractFactory
+  def get_color
+    raise "must implement getColor() method in subclass"
```

Comment:
Probably `getColor()` should be `get_color()` here to get Rubyism effect.
Same for `getShape()`.